### PR TITLE
Take password protection out of labs

### PR DESCRIPTION
**Pull request by acburdine**
refs #5073 

This takes password protection out of labs and moves it to Settings\General.

per @JohnONolan's request I added a random words generator library that generates random passwords when privacy is first enabled.

### CODE REVIEW

File: core/client/app/controllers/settings/general.js [hunk 1]
```diff
@@ -37,10 +38,24 @@ var SettingsGeneralController = Ember.Controller.extend({
...
+            }));
+        }
+    }),
+
     actions: {
         save: function () {
             var self = this;
 
+            if (this.get('model.isPrivate') && this.get('model.password') === '') {
+                self.notifications.showError('Password must have a value');
```
Comment (jaswilli) [inline core/client/app/controllers/settings/general.js]: This should probably be moved into the settings validator.

**Comment by jaswilli** [comment]
I'm not sure it's worth it to use an external dependency and attach ~20KB worth of random words to the global namespace.

In my opinion, I'd rather pick 50 or 100 words and put the code in `core/client/app/utils/`.

**Comment by acburdine** [comment]
fixed

**Comment by ErisDS** [comment]
:+1: for the 6 + 3 conbo as discussed in slack :) This seems to be working really nicely.

**Pull request merged**